internal/core/metadata/rest_deviceprofile.go

```diff
@@ -278,6 +281,29 @@ func restAddProfileByYaml(
 		return
 	}
 
+	if configuration.Writable.EnableValueDescriptorManagement {
+		// Check if the device profile name is unique so that we do not create ValueDescriptors for a DeviceProfile that
+		// will fail during the creation process later on.
+		nameOp := device_profile.NewGetProfileName(dp.Name, dbClient)
+		_, err := nameOp.Execute()
+		// The operator will return an ItemNotFound error if the DeviceProfile can not be found.
```

Comment:
Is the `DuplicateName` error concept supposed to be addressing this comment? If so, can we add some note that that is the case? If not, let's make sure we catch that.

Reply:
I modified the comment, it might be more appropriate.

> The operator will return an DuplicateName error if the DeviceProfile exist.